druntime/src/core/internal/array/capacity.d

```diff
@@ -9,62 +9,150 @@
 */
 module core.internal.array.capacity;
 
-// HACK: `nothrow` and `pure` is faked.
-private extern (C) void[] _d_arraysetlengthT(const TypeInfo ti, size_t newlength, void[]* p) nothrow pure;
-private extern (C) void[] _d_arraysetlengthiT(const TypeInfo ti, size_t newlength, void[]* p) nothrow pure;
-
 /*
- * This template is needed because there need to be a `_d_arraysetlengthTTrace!Tarr` instance for every
- * `_d_arraysetlengthT!Tarr`. By wrapping both of these functions inside of this template we force the
- * compiler to create a instance of both function for every type that is used.
+ * Fully templated implementation of `_d_arraysetlengthT`, removing reliance on `TypeInfo`.
+ * The old `_d_arraysetlengthT` and `_d_arraysetlengthiT` functions have been removed.
+ */
+
+/**
+ * Resize a dynamic array by setting the `.length` property.
+ *
+ * Newly created elements are initialized to their default value.
+ *
+ * This function is **now fully templated**, eliminating the need for TypeInfo-based
+ * `_d_arraysetlengthT` and `_d_arraysetlengthiT`, while efficiently handling both
+ * zero-initialized and custom-initialized types.
```

Comment:
Again, no need to reference the old hooks. Just copy and paste the ddoc comments from `rt/lifetime.d` and update the parameters.